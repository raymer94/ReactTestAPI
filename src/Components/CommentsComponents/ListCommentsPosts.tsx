import { CommentsModelsDTO } from "./Comments.models";
import * as ReactBootstrap from 'react-bootstrap';
import { Link } from "react-router-dom";

interface ListCommentsProps{
    model?: CommentsModelsDTO[],
    showModal: any
}

export default function ListCommentsPosts(props: ListCommentsProps){
    return(
        <>
            <h3>List of comments</h3>
            <Link className="btn btn-danger mb-2" to={"/"}>Back</Link>
            <button className="btn btn-primary mb-2 mx-sm-3" onClick={props.showModal}>Add new comment</button>
                <ReactBootstrap.Table striped bordered hover>
                    <thead className="">
                        <tr>
                            <th>postId</th>
                            <th>id</th>
                            <th>name</th>
                            <th>email</th>
                            <th>body</th>
                        </tr>
                    </thead>
                    <tbody>

                        {props.model?.map(val =>
                            <tr key={val.id}>
                                <td>{val.postId}</td>
                                <td>{val.id}</td>
                                <td>{val.name}</td>
                                <td>{val.email}</td>
                                <td>{val.body}</td>
                            </tr>
                        )}
                    </tbody>
                </ReactBootstrap.Table>
        </>
    )
}

