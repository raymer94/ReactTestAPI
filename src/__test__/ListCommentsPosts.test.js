import React from 'react';
import renderer from 'react-test-renderer';
import ListCommentsPosts from '../Components/CommentsComponents/ListCommentsPosts';
import { BrowserRouter } from "react-router-dom";

test('ListComments Snapshot test', () => {
  const component = renderer.create(
    <BrowserRouter>
      <ListCommentsPosts/>
    </BrowserRouter>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});