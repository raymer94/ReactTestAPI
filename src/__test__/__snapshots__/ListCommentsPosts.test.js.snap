// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
Array [
  <h3>
    List of comments
  </h3>,
  <a
    className="btn btn-danger mb-2"
    href="/"
    onClick={[Function]}
  >
    Back
  </a>,
  <button
    className="btn btn-primary mb-2 mx-sm-3"
  >
    Add new comment
  </button>,
  <table
    className="table table-striped table-bordered table-hover"
  >
    <thead
      className=""
    >
      <tr>
        <th>
          postId
        </th>
        <th>
          id
        </th>
        <th>
          name
        </th>
        <th>
          email
        </th>
        <th>
          body
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
]
`;

exports[`ListComments Snapshot test 1`] = `
Array [
  <h3>
    List of comments
  </h3>,
  <a
    className="btn btn-danger mb-2"
    href="/"
    onClick={[Function]}
  >
    Back
  </a>,
  <button
    className="btn btn-primary mb-2 mx-sm-3"
  >
    Add new comment
  </button>,
  <table
    className="table table-striped table-bordered table-hover"
  >
    <thead
      className=""
    >
      <tr>
        <th>
          postId
        </th>
        <th>
          id
        </th>
        <th>
          name
        </th>
        <th>
          email
        </th>
        <th>
          body
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
]
`;
